app/main.py

```diff
@@ -1,3 +1,42 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as file:
+        data = json.loads(file.read())
+
+    fuel_price = data["FUEL_PRICE"]
+    customers = [Customer.from_dict(person)
+                 for person in data["customers"]]
+    shops = [Shop.from_dict(shop)
+             for shop in data["shops"]]
+
+    for person in customers:
+        print(f"{person.name} has {person.money} dollars")
+        choices = {}
+        for shop in shops:
+            if shop.is_in_stock(person.product_cart):
+                check = shop.check(person.name,
+                                   person.product_cart)
+                choice = {"name": shop.name,
+                          "check": check[1]}
+                cost = check[0]
+                cost += person.car.fuel_cost(person.location,
+                                             shop.location,
+                                             fuel_price)
+                if person.money >= cost:
+                    choices[cost] = choice
```

Comment:
What happens costs of trips to several stores have the same cost?

Reply:
Choose last of shops with same profit